server/src/service.rs

```diff
@@ -15,12 +15,15 @@ use zbus::{
     zvariant::{ObjectPath, OwnedObjectPath, OwnedValue, Value},
 };
 
-use crate::{service_manager::ServiceManager, session::Session};
+use crate::{collection::Collection, service_manager::ServiceManager, session::Session};
 
 pub type Result<T> = std::result::Result<T, ServiceError>;
 
 #[derive(Debug)]
 pub struct Service {
+    // Secret Service properties
+    collections: Mutex<Vec<OwnedObjectPath>>,
```

Comment:
Is this correct? How are you going to get the collection again from it path?

Reply:
> Is this correct?

Yes. I'm certain for single collection/keyring. But, we may need to modify this when adding multiple keyring support. I did this to avoid many instances of cloning.

> How are you going to get the collection again from it path?

I can access all the methods and attributes following [this](https://docs.rs/zbus/latest/zbus/object_server/struct.ObjectServer.html#examples) approach.